tests/API/Management/JobsTest.php

```diff
@@ -0,0 +1,228 @@
+        $this->assertStringStartsWith( 'multipart/form-data', $headers['Content-Type'][0] );
+
+        $form_body     = $api->getHistoryBodyAsString();
+        $form_body_arr = explode( "\r\n", $form_body );
+
+        // Test that the form data contains our import file content.
+        $import_content = file_get_contents( self::TEST_IMPORT_USERS_JSON_PATH );
+        $this->assertContains( 'name="users"; filename="test-import-users-file.json"', $form_body );
+        $this->assertContains( $import_content, $form_body );
+
+        $conn_id_key = array_search( 'Content-Disposition: form-data; name="connection_id"', $form_body_arr );
+        $this->assertNotEmpty( $conn_id_key );
+        $this->assertEquals( '__test_conn_id__', $form_body_arr[$conn_id_key + self::FORM_DATA_VALUE_KEY_OFFSET] );
+
+        $upsert_key = array_search( 'Content-Disposition: form-data; name="upsert"', $form_body_arr );
+        $this->assertNotEmpty( $upsert_key );
+        $this->assertEquals( 'true', $form_body_arr[$upsert_key + self::FORM_DATA_VALUE_KEY_OFFSET] );
+
+        $email_key = array_search( 'Content-Disposition: form-data; name="send_completion_email"', $form_body_arr );
+        $this->assertNotEmpty( $email_key );
+        $this->assertEquals( 'true', $form_body_arr[$email_key + self::FORM_DATA_VALUE_KEY_OFFSET] );
+
+        $ext_id_key = array_search( 'Content-Disposition: form-data; name="external_id"', $form_body_arr );
+        $this->assertNotEmpty( $ext_id_key );
+        $this->assertEquals( '__test_ext_id__', $form_body_arr[$ext_id_key + self::FORM_DATA_VALUE_KEY_OFFSET] );
+    }
+
+
+    /**
+     * @throws \Exception Should not be thrown in this test.
+     */
+    public function testThatSendVerificationEmailIsFormedProperly()
+    {
+        $api = new MockManagementApi( [ new Response( 200, self::$headers ) ] );
+
+        $api->call()->jobs->sendVerificationEmail( '__test_user_id__' );
+
+        $this->assertEquals( 'POST', $api->getHistoryMethod() );
+        $this->assertEquals( 'https://api.test.local/api/v2/jobs/verification-email', $api->getHistoryUrl() );
+        $this->assertEmpty( $api->getHistoryQuery() );
+
+        $body = $api->getHistoryBody();
+        $this->assertArrayHasKey( 'user_id', $body );
+        $this->assertEquals( '__test_user_id__', $body['user_id'] );
+
+        $headers = $api->getHistoryHeaders();
+        $this->assertEquals( 'Bearer __api_token__', $headers['Authorization'][0] );
+        $this->assertEquals( self::$expectedTelemetry, $headers['Auth0-Client'][0] );
+        $this->assertEquals( 'application/json', $headers['Content-Type'][0] );
+    }
+
+    /**
+     * @throws \Auth0\SDK\Exception\ApiException
+     * @throws \Exception
+     */
+    public function testIntegrationImportUsersJob()
+    {
+        $env = self::getEnv();
+
+        if (! $env['API_TOKEN']) {
+            $this->markTestSkipped( 'No client secret; integration test skipped' );
```

Comment:
can this "integration tests" suite be extracted to separate classes? The idea I have in mind is to run them with a different command (e.g. passing a different regex to the test files to include and run) so that CI script can tell when the ENV is available and run one test suite or the other (or both if that's the case)

Reply:
This works well for now.